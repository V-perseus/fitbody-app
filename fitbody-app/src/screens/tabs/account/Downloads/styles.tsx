import { Platform, StatusBar, StyleSheet } from 'react-native'
import globals from '../../../../config/globals'

const styles = StyleSheet.create({
  container: {
    ...globals.styles.container,
    alignItems: 'center',
    backgroundColor: globals.styles.colors.colorWhite,
  },
  view: {
    ...globals.styles.view,
    backgroundColor: globals.styles.colors.colorWhite,
    width: '100%',
    marginTop: Platform.OS === 'ios' ? 50 : 50 + (StatusBar.currentHeight || 0),
  },
  headerTitle: {
    ...globals.header.headerTitleStyle,
  },
  notificationsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingRight: 15,
  },
  buttonText: {
    color: globals.styles.colors.colorBlackDark,
  },
  footnote: {
    fontSize: 10,
    fontStyle: 'italic',
    color: globals.styles.colors.colorGrayDark,
  },
  noDownloads: {
    fontFamily: globals.fonts.primary.style.fontFamily,
    fontSize: 15,
    letterSpacing: -0.3,
    textAlign: 'center',
    color: globals.styles.colors.colorBlack,
    marginTop: 13,
  },
  title: {
    fontFamily: globals.fonts.primary.style.fontFamily,
    fontSize: 15,
    letterSpacing: -0.3,
    color: globals.styles.colors.colorBlack,
  },
  subTitle: {
    fontFamily: globals.fonts.secondary.style.fontFamily,
    fontSize: 20,
    color: globals.styles.colors.colorBlack,
  },
  downloadItemContainer: {
    paddingHorizontal: 24,
    paddingVertical: 14,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderBottomColor: globals.styles.colors.colorGray,
    borderBottomWidth: 1,
  },
})

export default styles
