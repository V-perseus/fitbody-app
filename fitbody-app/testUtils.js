import React from 'react'
import { render as rtlRender } from '@testing-library/react-native'
import { configureStore } from '@reduxjs/toolkit'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { Provider } from 'react-redux'

// Import your own reducer
import { appReducer } from './src/store'
import { ModalStackScreens } from './src/config/routes/loggedInStack'

// custom render method override
function render(ui, { preloadedState, store = configureStore({ reducer: appReducer, preloadedState }), ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react-native'
// override render method
export { render }

// custom navigation mock
const Stack = createStackNavigator()
export const MockedNavigator = ({
  component,
  params = {},
  preloadedState,
  store = configureStore({ reducer: appReducer, preloadedState }),
}) => {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="MockedScreen" component={component} initialParams={params} />
          <Stack.Screen name="Modals" component={ModalStackScreens} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  )
}
