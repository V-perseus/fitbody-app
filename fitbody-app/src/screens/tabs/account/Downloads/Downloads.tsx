import React, { useState, useEffect, useMemo, useLayoutEffect } from 'react'
import { Text, StatusBar, View, FlatList } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'
import RNFS from 'react-native-fs'

// Assets
import styles from './styles'
import globals from '../../../../config/globals'

// Components
import DownloadsIcon from '../../../../../assets/images/svg/icon/40px/circle/downloaded.svg'
import CloseCircle from '../../../../../assets/images/svg/icon/16px/circle/close-circle.svg'
import { ButtonOpacity } from '../../../../components/Buttons/ButtonOpacity'
import { HeaderButton } from '../../../../components/Buttons/HeaderButton'

// Services
import { clearChallengeMonth, clearDownloads } from '../../../../data/workout'
import { useAppSelector } from '../../../../store/hooks'
import { IProgram, ITrainer } from '../../../../data/workout/types'

// Types
import { DownloadsScreenNavigationProps } from '../../../../config/routes/routeTypes'

type DownloadFileData = {
  program: IProgram
  trainer: ITrainer
  size: number
}

type ChallengeDownloadFileData = {
  title: string
  color: string
  size: number
}

interface IDownloadsProps {
  navigation: DownloadsScreenNavigationProps
}
const Downloads: React.FC<IDownloadsProps> = ({ navigation }) => {
  const progs = useAppSelector((state) => state.data.workouts.programs)
  const programs = useMemo(() => Object.values(progs), [progs])

  const trainersObj = useAppSelector((state) => state.data.workouts.trainers)
  const trainers = useMemo(() => Object.values(trainersObj), [trainersObj])

  const cpS = useAppSelector((state) => state.data.user.workout_goal)
  const currentProgram = useMemo(() => (programs ? programs.find((p) => p.slug === cpS) : null), [programs, cpS])

  const [downloads, setDownloads] = useState<DownloadFileData[]>([])
  const [challengeDownload, setChallengeDownload] = useState<ChallengeDownloadFileData>()

  useLayoutEffect(() => {
    navigation.setOptions({
      headerTitle: () => <Text style={styles.headerTitle}>DOWNLOADS</Text>,
      headerLeft: () => <HeaderButton onPress={() => navigation.goBack()} />,
    })
  }, [navigation])

  useEffect(() => {
    const enumerate = async () => {
      const pathC = `${RNFS.DocumentDirectoryPath}/assets/challenges`
      const foldersC = await RNFS.readDir(pathC)
      const size = foldersC?.filter((f) => f.isFile()).reduce((total, curr) => total + parseInt(curr.size, 10), 0) / (1024 * 1024)
      if (size) {
        setChallengeDownload({
          title: '5 MINUTE CARDIO BURN',
          color: globals.styles.colors.colorBlack,
          size,
        })
      }
    }
    enumerate()
  }, [])

  useEffect(() => {
    const enumerate = async () => {
      try {
        const path = `${RNFS.DocumentDirectoryPath}/assets/workouts`
        const folders = await RNFS.readDir(path)

        const result = await Promise.all(
          folders
            .filter((f) => f.isDirectory())
            .map(async (ff) => {
              const files = await RNFS.readDir(ff.path)
              // console.log('name', ff.name)
              const size = files.reduce((tot, cur) => tot + parseInt(cur.size, 10), 0)
              return {
                program: programs.find((p) => p.id === parseInt(ff.name, 10))!,
                trainer: trainers.find((t) => t.programs.includes(parseInt(ff.name, 10)))!,
                size: parseFloat(String(size / 1000000)),
              }
            }),
        )
        // console.log('RES', result)

        setDownloads(
          result.sort((a, b) => a.program.sort_order - b.program.sort_order).sort((a, b) => a.trainer.sort_order - b.trainer.sort_order),
        )
      } catch (error) {
        // most likely the directory doesn't exist yet when running RNFS.readDir. just catch the rejection and do nothing
        // as this can happen if a user hadn't downloded any workouts before visiting this screen
      }
    }

    enumerate()
  }, [programs, trainers])

  const clearProgram = (id: number) => () => {
    const path = `${RNFS.DocumentDirectoryPath}/assets/workouts/${id}`
    RNFS.unlink(path)
    setDownloads((d) => d.filter((p) => p.program.id !== id))
    clearDownloads({ programId: id })
  }

  function handleDownloadItemPress(item: DownloadFileData) {
    if (item.program.id !== currentProgram?.id) {
      navigation.navigate('Modals', {
        screen: 'ConfirmationDialog',
        params: {
          yesLabel: 'CLEAR',
          noLabel: 'NEVER MIND',
          yesHandler: clearProgram(item.program.id),
          title: 'Are you sure?',
          body: 'You are about to clear all downloads for this program. This will not clear any workout history.',
        },
      })
    }
  }

  const clearChallengeProgram = () => () => {
    // this simply triggers a redownload of all monthly challenge/cardio burn data
    const path = `${RNFS.DocumentDirectoryPath}/assets/challenges`
    RNFS.unlink(path)
    clearChallengeMonth()
    setChallengeDownload(undefined)
  }

  function handleChallengeDownloadItemPress() {
    navigation.navigate('Modals', {
      screen: 'ConfirmationDialog',
      params: {
        yesLabel: 'CLEAR',
        noLabel: 'NEVER MIND',
        yesHandler: clearChallengeProgram(),
        title: 'Are you sure?',
        body: 'You are about to restore all downloads for this months 5 Minute Cardio Burn. This will not clear any workout history.',
      },
    })
  }

  function renderItem({ item, index }: { item: DownloadFileData; index: number }) {
    return (
      <View key={index} style={styles.downloadItemContainer}>
        <Text style={{ color: item.trainer.color, fontFamily: globals.fonts.secondary.style.fontFamily, fontSize: 20 }}>
          {item.program.title}
        </Text>
        {/* <SvgUri uri={resolveLocalUrl(item.program.logo_small_url)} color={item.trainer.color} /> */}
        <View style={{ flexDirection: 'row', alignItems: 'center' }}>
          <Text
            style={{
              fontFamily: globals.fonts.primary.semiboldStyle.fontFamily,
              fontSize: 12,
              color: globals.styles.colors.colorGrayDark,
              marginRight: 16,
            }}>
            {Math.round(item.size)} MB
          </Text>
          <ButtonOpacity onPress={() => handleDownloadItemPress(item)} testID={item.program.title}>
            <View>
              <DownloadsIcon
                color={item.program.id === currentProgram?.id ? globals.styles.colors.colorGray : globals.styles.colors.colorSkyBlue}
                style={{ margin: 2 }}
              />
              <CloseCircle
                color={item.program.id === currentProgram?.id ? globals.styles.colors.colorGrayDark : globals.styles.colors.colorBlackDark}
                style={{ position: 'absolute', top: 0, right: 0 }}
              />
            </View>
          </ButtonOpacity>
        </View>
      </View>
    )
  }

  return (
    <SafeAreaView style={styles.container}>
      <View style={[styles.view, { flex: 0 }]}>
        <StatusBar barStyle="dark-content" />
        {/* <View style={{ flexDirection: 'row', margin: 24, alignItems: 'center', justifyContent: 'space-between' }}>
          <Text style={{ fontFamily: globals.fonts.primary.boldStyle.fontFamily, fontSize: ms(14) }}>Automatically Download Over Wifi Only</Text>
          <Switch
            thumbColor={'#fff'}
            disabled={isPaused && indent !== 0}
            value={indent === 0 ? !isEnabled : isEnabled}
            onValueChange={(v) => toggleSetting(setting, indent === 0 ? !v : v)}
            trackColor={{ true: '#ff6c8f' }}
          />
        </View> */}
        <View style={{ marginHorizontal: 24, marginBottom: 22 }}>
          {/* <Text style={{ fontFamily: globals.fonts.primary.style.fontFamily, fontSize: 12 }}>
            When toggled on, your workouts will only be downloaded automatically when connected to Wi-Fi.{'\n\n'}When toggled off, your
            workouts will download automatically using cellular data.
          </Text> */}
          <Text style={{ fontFamily: globals.fonts.primary.style.fontFamily, fontSize: 12 }}>
            {'\n'}
            Please note you cannot delete data for the workout program you currently have open.
          </Text>
        </View>
        <View style={{ backgroundColor: globals.styles.colors.colorGrayLight, alignSelf: 'stretch', alignItems: 'center', padding: 5 }}>
          <Text
            style={{ fontFamily: globals.fonts.primary.semiboldStyle.fontFamily, color: globals.styles.colors.colorBlack, fontSize: 14 }}>
            Downloads
          </Text>
        </View>
        <FlatList data={downloads} keyExtractor={(item, index) => `download_${index}`} renderItem={renderItem} />
        {challengeDownload && (
          <View style={styles.downloadItemContainer}>
            <Text style={{ color: challengeDownload.color, fontFamily: globals.fonts.secondary.style.fontFamily, fontSize: 20 }}>
              {challengeDownload.title}
            </Text>
            {/* <SvgUri uri={resolveLocalUrl(item.program.logo_small_url)} color={item.trainer.color} /> */}
            <View style={{ flexDirection: 'row', alignItems: 'center' }}>
              <Text
                style={{
                  fontFamily: globals.fonts.primary.semiboldStyle.fontFamily,
                  fontSize: 12,
                  color: globals.styles.colors.colorGrayDark,
                  marginRight: 16,
                }}>
                {Math.round(challengeDownload.size)} MB
              </Text>
              <ButtonOpacity onPress={handleChallengeDownloadItemPress} testID={challengeDownload.title}>
                <View>
                  <DownloadsIcon color={globals.styles.colors.colorSkyBlue} style={{ margin: 2 }} />
                  <CloseCircle color={globals.styles.colors.colorBlackDark} style={{ position: 'absolute', top: 0, right: 0 }} />
                </View>
              </ButtonOpacity>
            </View>
          </View>
        )}
      </View>
    </SafeAreaView>
  )
}

export default Downloads
