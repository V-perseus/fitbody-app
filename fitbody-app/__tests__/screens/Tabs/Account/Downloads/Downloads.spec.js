import { MockedNavigator, render, act, cleanup, fireEvent } from '../../../../../testUtils'
import RNFS from 'react-native-fs'

import Downloads from '../../../../../src/screens/tabs/account/Downloads'

describe('Downloads screen', () => {
  let component = null
  beforeEach(() => {
    component = MockedNavigator({
      component: Downloads,
      preloadedState: {
        data: {
          user: {
            workout_goal: 'TONE',
          },
          workouts: {
            programs: {
              SHRED: { id: 1, title: 'Shred', sort_order: 1, slug: 'SHRED' },
              TONE: { id: 2, title: 'Tone', sort_order: 2, slug: 'TONE' },
            },
            trainers: [
              {
                programs: [2],
                sort_order: 1,
              },
              {
                programs: [1],
                sort_order: 2,
              },
            ],
          },
        },
      },
    })
  })
  afterEach(() => {
    cleanup()
  })

  it('renders snapshot correctly', async () => {
    const tree = render(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render header', async () => {
    const { findByText } = render(component)
    await findByText('DOWNLOADS')
    await findByText(/Please note you cannot delete data for the workout program you currently have open/gi)
  })

  it('should respond with mocked file data', async () => {
    RNFS.readDir.mockResolvedValue([
      {
        ctime: '2022-05-11T16:06:51.804Z',
        mtime: '2022-05-11T16:07:14.512Z',
        name: 2,
        path: '/var/mobile/Containers/Data/Application/6A864251-F3D2-433C-ACB1-91D3A0A891A2/Documents/assets/workouts/2',
        size: 8032,
        isFile: jest.fn(),
        isDirectory: () => true,
      },
    ])
    // .mockResolvedValueOnce([
    //   {
    //     ctime: '2022-05-11T16:07:14.298Z',
    //     mtime: '2022-05-11T23:07:06.000Z',
    //     name: 'kYpeVQ2Xmm0rlqTUERGTI70LhDuRE2kdPiNGxnVJ.mp4',
    //     path: '/var/mobile/Containers/Data/Application/6A864251-F3D2-433C-ACB1-91D3A0A891A2/Documents/assets/workouts/2/kYpeVQ2Xmm0rlqTUERGTI70LhDuRE2kdPiNGxnVJ.mp4',
    //     size: 90825,
    //     isFile: jest.fn(),
    //     isDirectory: jest.fn(),
    //   },
    // ])
    // const tree = render(component)
    const { findByText } = render(component)
    await act(async () => {
      await findByText(/TONE/i)
    })
  })

  it('should nav to delete warning modal', async () => {
    RNFS.readDir.mockResolvedValue([
      {
        ctime: '2022-05-11T16:06:51.804Z',
        mtime: '2022-05-11T16:07:14.512Z',
        name: 2,
        path: '/var/mobile/Containers/Data/Application/6A864251-F3D2-433C-ACB1-91D3A0A891A2/Documents/assets/workouts/2',
        size: 8032,
        isFile: jest.fn(),
        isDirectory: () => true,
      },
      {
        ctime: '2022-05-11T16:06:51.804Z',
        mtime: '2022-05-11T16:07:14.512Z',
        name: 1,
        path: '/var/mobile/Containers/Data/Application/6A864251-F3D2-433C-ACB1-91D3A0A891A2/Documents/assets/workouts/1',
        size: 8032,
        isFile: jest.fn(),
        isDirectory: () => true,
      },
    ])
    const { findByTestId, findByText } = render(component)
    await act(async () => {
      const deleteBtn = await findByTestId(/Shred/i)
      await fireEvent.press(deleteBtn)
      const modalBody = await findByText('You are about to clear all downloads for this program. This will not clear any workout history.')
      expect(modalBody).toBeTruthy()
    })
  })
})
