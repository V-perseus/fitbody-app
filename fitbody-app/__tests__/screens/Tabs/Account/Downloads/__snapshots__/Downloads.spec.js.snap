// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Downloads screen renders snapshot correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  hitSlop={16}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "opacity": 1,
                      },
                      Object {
                        "alignItems": "center",
                        "borderRadius": 30,
                        "display": "flex",
                        "height": 42,
                        "justifyContent": "center",
                        "marginHorizontal": 7,
                        "width": 42,
                      },
                      undefined,
                    ]
                  }
                >
                  <arrow-left.svg
                    color="#333333"
                    testID="back_arrow"
                  />
                </View>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 590,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontFamily": "OpenSans-Semibold",
                      "fontSize": 16,
                      "textAlign": "center",
                    }
                  }
                >
                  DOWNLOADS
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RNCSafeAreaView
                        edges={
                          Array [
                            "bottom",
                            "left",
                            "right",
                            "top",
                          ]
                        }
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#ffffff",
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#ffffff",
                                "flex": 1,
                                "flexDirection": "column",
                                "marginHorizontal": 15,
                                "marginTop": 50,
                                "width": "100%",
                              },
                              Object {
                                "flex": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "marginBottom": 22,
                                "marginHorizontal": 24,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "fontFamily": "OpenSans",
                                  "fontSize": 12,
                                }
                              }
                            >
                              

                              Please note you cannot delete data for the workout program you currently have open.
                            </Text>
                          </View>
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "alignSelf": "stretch",
                                "backgroundColor": "#f3f3f3",
                                "padding": 5,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#333333",
                                  "fontFamily": "OpenSans-SemiBold",
                                  "fontSize": 14,
                                }
                              }
                            >
                              Downloads
                            </Text>
                          </View>
                          <RCTScrollView
                            data={Array []}
                            getItem={[Function]}
                            getItemCount={[Function]}
                            keyExtractor={[Function]}
                            onContentSizeChange={[Function]}
                            onLayout={[Function]}
                            onMomentumScrollBegin={[Function]}
                            onMomentumScrollEnd={[Function]}
                            onScroll={[Function]}
                            onScrollBeginDrag={[Function]}
                            onScrollEndDrag={[Function]}
                            removeClippedSubviews={false}
                            renderItem={[Function]}
                            scrollEventThrottle={50}
                            stickyHeaderIndices={Array []}
                            viewabilityConfigCallbackPairs={Array []}
                          >
                            <View />
                          </RCTScrollView>
                        </View>
                      </RNCSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
